src/main/kotlin/kotlinx/kover/gradle/plugin/appliers/ProjectApplier.kt

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2017-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.kover.gradle.plugin.appliers
+
+import kotlinx.kover.gradle.plugin.commons.*
+import kotlinx.kover.gradle.plugin.dsl.*
+import kotlinx.kover.gradle.plugin.dsl.internal.*
+import kotlinx.kover.gradle.plugin.locators.*
+import kotlinx.kover.gradle.plugin.tasks.internal.KoverAgentJarTask
+import kotlinx.kover.gradle.plugin.tasks.internal.KoverArtifactGenerationTask
+import kotlinx.kover.gradle.plugin.tools.*
+import kotlinx.kover.gradle.plugin.tools.CoverageToolFactory
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.kotlin.dsl.*
+
+/**
+ * Main Gradle Plugin applier of the project.
+ */
+internal class ProjectApplier(private val project: Project) {
+    private lateinit var projectExtension: KoverProjectExtensionImpl
+    private lateinit var androidExtension: KoverAndroidExtensionImpl
+    private lateinit var regularReportExtension: KoverReportExtensionImpl
+
+    /**
+     * The code executed right at the moment of applying of the plugin.
+     */
+    fun onApply() {
+        project.configurations.create(DEPENDENCY_CONFIGURATION_NAME) {
+            asBucket()
+        }
+
+        projectExtension = project.extensions.create(PROJECT_EXTENSION_NAME, project.objects)
+        regularReportExtension = project.extensions.create(REGULAR_REPORT_EXTENSION_NAME)
+        androidExtension = project.extensions.create(ANDROID_EXTENSION_NAME, project.objects)
```

Comment:
Yes, because Gradle allows any order of application of plugins and we cannot be sure that after the Kover plugin, the Android plugin will not be applied